### PR TITLE
Remove errors and warning for local packages without a library component

### PR DESCRIPTION
**Pull request by kmicklas**
Currently `ob shell` and derivative commands will print errors like this if you have a local package without a library component:

```
Haskell package has no library component
Failed to find buildable packages in $path_to_cabal_file
```

I don't see any reason why this should be considered an erroneous case.

This removes the `Failed to find buildable packages in ...` warning, but I don't believe it's necessary since any actual errors or warnings will still be reported by `parseCabalPackage`.

I have:

  - [x] Based work on latest `develop` branch
  - [ ] Looked for lint in my changes with `hlint .` (lint found code you did not write can be left alone)
  - [ ] Run the test suite: `$(nix-build -A selftest --no-out-link)`
  - [ ] (Optional) Run CI tests locally: `nix-build release.nix -A build.x86_64-linux --no-out-link` (or `x86_64-darwin` on macOS)

### CODE REVIEW

**Comment by kmicklas** [comment]
@3noch Anything blocking merging this?

File: lib/command/src/Obelisk/Command/Run.hs [hunk 1]
```diff
@@ -386,9 +387,6 @@ parsePackagesOrFail dirs' = do
       "No valid, buildable packages found" <> (if null dirs then "" else " in " <> intercalate ", " dirs)
     Just xs -> pure xs
 
-  unless (null pkgDirErrs) $
-    putLog Warning $ T.pack $ "Failed to find buildable packages in " <> intercalate ", " pkgDirErrs
```
Comment (Ericson2314) [inline lib/command/src/Obelisk/Command/Run.hs:390]: Is this change related?

Comment (kmicklas) [reply lib/command/src/Obelisk/Command/Run.hs:390]: It's related in that you also get this warning for a package with no library component, in addition to the error above. I don't think either of them should be considered out of the ordinary.

Comment (Ericson2314) [reply lib/command/src/Obelisk/Command/Run.hs:390]: Ah and there is no more `Either`.

**Review by Ericson2314 — APPROVED** [review]

**Comment by ali-abrar** [comment]
Thanks, Kenny
